engine/access/rest/request/height.go

```diff
@@ -0,0 +1,75 @@
+package request
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+)
+
+const sealed = "sealed"
+const final = "final"
+
+// Special height values
+const SealedHeight uint64 = math.MaxUint64 - 1
+const FinalHeight uint64 = math.MaxUint64 - 2
+const EmptyHeight uint64 = math.MaxUint64 - 3
+
+type Height uint64
+
+func (h *Height) Parse(raw string) error {
+	if raw == "" { // allow empty
+		*h = Height(EmptyHeight)
+		return nil
+	}
+
+	if raw == sealed {
+		*h = Height(SealedHeight)
+		return nil
+	}
+	if raw == final {
+		*h = Height(FinalHeight)
+		return nil
+	}
+
+	height, err := strconv.ParseUint(raw, 0, 64)
+	if err != nil {
+		return fmt.Errorf("invalid height format")
+	}
+
+	*h = Height(height)
```

Comment:
since we're overloading the top 3 `uint64` values, we should also check that height is less than `EmptyHeight`

Reply:
good point